src/main/drivers/bus_i2c.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *           
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *  
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ */ 
+
+#include <stdbool.h>
+#include <stdint.h> 
+#include <stdlib.h>
+    
+#include <platform.h>
+
+#ifdef USE_I2C
+
+#include "io.h"
+#include "system.h"
+
+#include "config/config_master.h"
+
+#include "bus_i2c.h"
+#include "nvic.h"
+#include "io_impl.h"
+#include "rcc.h"
+
+/*
+ * XXX To pursue future (further) per-MCU config scheme:
+ * https://github.com/betaflight/betaflight/pull/1867#discussion_r93647427
+ * (@ledvinap) This way to find I2C device can't handle case when multiple
+ * driver instances share identical pins. It is (probably) not problem for I2C,
+ * but for example F3 ADC can connect single pin to multiple ADCs.
+ * It would be nice to use common mechanism to specify pins for all drivers ...
+ */
+
+static i2cDevice_t *i2cConfigFindMap(ioTag_t scl, ioTag_t sda)
+{
+    for (size_t map = 0 ; map < i2cPinMapSize() ; map++) {
+        if ((scl == i2cPinMap[map].scl) && (sda == i2cPinMap[map].sda))
+            return &i2cPinMap[map];
+    }
+
+    return NULL;
+}
+
+void i2cPinConfigSet(i2cTargetConfig_t *pTargetConfig)
```

Comment:
Define this as `static void i2cPinConfigSet(i2cPinConfig_t *i2cPinConfig, const i2cTargetConfig_t *pTargetConfig)`